core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -74,7 +74,7 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
   val pollTask: Cancellable =
     context.system.scheduler.schedule(pollInterval(), pollInterval(), self, Poll)(context.dispatcher)
 
-  var requests = Map.empty[TopicPartition, ActorRef]
+  var requests = Map.empty[ActorRef, RequestMessages]
```

Comment:
yeah, then I guess I should rewrite to java.util.HashMap and while loops. There are plenty of collection juggling in this actor that can be improved. Not sure I want to do that now. I'll take a look.